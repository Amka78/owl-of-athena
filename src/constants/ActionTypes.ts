const LOGIN = "LOGIN" as const;
const UPDATE_USER = "UPDATE_USER" as const;
const LOGOUT = "LOGOUT" as const;
const CACHE_SETTINGS = "CACHE_SETTINGS" as const;
const INIT = "INIT" as const;
const CACHE_PROFILES = "CACHE_PROFILES" as const;
const CACHE_SESSIONS = "CACHE_SESSIONS" as const;
const SELECT_SESSION = "SELECT_SESSION" as const;
const SELECT_SESSION_DETAIL = "SELECT_SESSION_DETAIL" as const;
const INITIALIZE_AURORA = "INITIALIZE_AURORA" as const;
const INITIALIZE_SESSION = "INITIALIZE_SESSION" as const;
const UPDATE_FILTER = "UPDATE_FILTER" as const;
const DELETE_SESSION = "DELETE_SESSION" as const;
const UPDATE_SESSION = "UPDATE_SESSION" as const;
export default {
    INIT,
    INITIALIZE_AURORA,
    INITIALIZE_SESSION,
    LOGIN,
    LOGOUT,
    UPDATE_USER,
    CACHE_SESSIONS,
    DELETE_SESSION,
    SELECT_SESSION,
    SELECT_SESSION_DETAIL,
    UPDATE_FILTER,
    UPDATE_SESSION,
    CACHE_SETTINGS,
    CACHE_PROFILES,
};
