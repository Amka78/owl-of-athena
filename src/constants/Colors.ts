const teal = "#00ffe6";
const purple = "#4a2fba";
const white = "#ffffff";
export default {
    black: "#000000",
    charcoal: "#333333",
    gray: "#9B9B9B",
    white: white,
    purple: purple,
    navy: "#070724",
    navy_darker: "#09092B",
    blue: "#001438",
    cyan: "#bcfeff",
    teal: teal,
    yellow: "#daf392",
    red: "#ff4310",
    light_orange: "#ff8260",
    semi_transparent: "#66000000",
    aurora_connected: teal,
    aurora_connected_text: purple,
    aurora_disconnected: purple,
    aurora_disconnected_text: white
};
