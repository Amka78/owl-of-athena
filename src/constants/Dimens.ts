export default {
    toolbar_title_text_size: 14,
    toolbar_button_text_size: 14,
    toolbar_button_margin_horizontal: 8,

    view_padding_bottom: 12,

    content_title_text_size: 24,
    content_title_margin_top: 32,
    content_title_margin_bottom: 48,
    content_text_size: 16,
    content_margin_horizontal: 48,

    snackbar_text_size: 8,

    statusbar_height: 8,

    navbar_height: 64,

    tip_text_size: 13,
    tip_text_margin_bottom: 13,

    error_text_size: 13,
    error_text_margin_bottom: 18,

    button_text_size: 24,
    button_height: 50,
    button_margin_bottom: 16,
    button_radius: 16,
    button_flat_text_size: 14,
    button_flat_margin_bottom: 12,
    button_flat_height: 28,

    input_text_size: 12,
    input_margin_bottom: 16,

    checkbox_text_size: 12,
    checkbox_outer_diameter: 24,
    checkbox_inner_diameter: 16,
    checkbox_gap: 4,
    checkbox_offset: 2,
    checkbox_padding_left: 16,
    checkbox_padding_right: 24,
    checkbox_margin_top: 16,
    checkbox_margin_bottom: 8,

    option_text_size: 14,
    option_height: 20,
    option_margin_bottom: 12,

    time_picker_height: 150,
    time_picker_scale: 1,
    time_picker_padding_horizontal: 0,

    circular_progress_diameter: 96,
    circular_progress_text_size: 32,

    // home fragment dimensions
    home_alarm_time_text_size: 48,
    home_alarm_meridian_text_size: 22,

    // session fragment dimensions
    session_label_text_size: 20,
    session_time_text_size: 28,
    session_sleep_score_text_size: 48,
    session_notes_text_size: 20,

    // session list fragment dimensions
    session_list_item_padding: 12,

    session_list_section_height: 24,
    session_list_section_padding_left: 24,
    session_list_section_text_size: 16,

    session_list_item_title_text_size: 20,
    session_list_item_subtitle_text_size: 15,
    session_list_item_text_margin_left: 18
};
