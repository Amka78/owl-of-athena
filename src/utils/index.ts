import BaseUrl from "./BaseUrl";
export { BaseUrl };
