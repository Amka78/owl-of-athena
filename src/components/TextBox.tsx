//#region Import Modules
import React, { FunctionComponent } from "react";
import { TextInputProps } from "react-native";
import { TextInput } from "react-native-paper";

import { Colors, Dimens, Fonts, Layout } from "../constants";
//#endregion

//#region Types
export type TextBoxProps = TextInputProps & {
    error?: boolean;
    label?: string;
};
//#endregion

//#region Component
export const TextBox: FunctionComponent<TextBoxProps> = (
    props: TextBoxProps
) => {
    return (
        <TextInput
            {...props}
            mode={"flat"}
            selectionColor={Colors.white}
            style={textInputStyle}
            underlineColor={Colors.white}
            theme={{
                colors: {
                    text: Colors.cyan,
                    placeholder: Colors.white,
                    primary: Colors.cyan,
                },
            }}
        ></TextInput>
    );
};
//#endregion

//#region Styles
const textInputStyle = {
    backgroundColor: "transparent",
    marginBottom: Dimens.button_margin_bottom,
    fontFamily: Fonts.primaryRegular,
    fontSize: Dimens.input_text_size,
    width: Layout.window.fixedWidth - Dimens.content_margin_horizontal * 2,
};
//#endregion
