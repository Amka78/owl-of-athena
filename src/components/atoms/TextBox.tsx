//#region Import Modules
import React, { FunctionComponent } from "react";
import { TextInputProps } from "react-native";
import { TextInput, useTheme } from "react-native-paper";

import { Colors, Dimens, Fonts, Layout } from "../../constants";
import { useWindowDimensions } from "../../hooks";
//#endregion

//#region Types
export type TextBoxProps = Omit<TextInputProps, "selectionColor"> & {
    error?: boolean;
    label?: string;
    selectionColor?: string;
    underlineColor?: string;
};
//#endregion

//#region Component
export const TextBox: FunctionComponent<TextBoxProps> = (
    props: TextBoxProps
) => {
    const theme = useTheme();
    const dimens = useWindowDimensions();
    let width = dimens.width - Dimens.content_margin_horizontal * 2;

    if (width > Dimens.input_text_max_width) {
        width = Dimens.input_text_max_width;
    }
    return (
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        //@ts-ignore
        <TextInput
            {...props}
            mode={"flat"}
            selectionColor={
                props.selectionColor ? props.selectionColor : Colors.white
            }
            style={[textInputStyle, { width }, props.style]}
            underlineColor={
                props.underlineColor ? props.underlineColor : Colors.white
            }
            theme={theme}
        ></TextInput>
    );
};
//#endregion

//#region Styles
const textInputStyle = {
    marginBottom: Dimens.button_margin_bottom,
    fontFamily: Fonts.primaryRegular,
    fontSize: Dimens.input_text_size,
    width: Layout.window.fixedWidth - Dimens.content_margin_horizontal * 2,
};
//#endregion
