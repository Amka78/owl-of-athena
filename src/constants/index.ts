import ActionTypes from "./ActionTypes";
import Colors from "./Colors";
import Dimens from "./Dimens";
import Fonts from "./Fonts";
import Layout from "./Layout";
import Message, { MessageLocalizationParam } from "./Message";
import { MessageKeys } from "./MessageKeys";
import StorageKeys from "./StorageKeys";

export {
    ActionTypes,
    Colors,
    Dimens,
    Fonts,
    Layout,
    MessageLocalizationParam,
    Message,
    MessageKeys,
    StorageKeys
};
