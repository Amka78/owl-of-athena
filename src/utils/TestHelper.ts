import "../components";

import { shallow, ShallowWrapper } from "enzyme";
import enzymetoJson from "enzyme-to-json";
import {
    GestureResponderEvent,
    NativeSyntheticEvent,
    TextInputEndEditingEventData,
} from "react-native";

import { ButtonProps } from "../components/atoms/Button";

// @ts-ignore
export const FakeGestureResponderEvent: GestureResponderEvent = {};
// @ts-ignore
export const FakeTextBoxOnEndEditingEventData: NativeSyntheticEvent<TextInputEndEditingEventData> = {};

export function createMock<P, S>(comp: any): ShallowWrapper<P, S> {
    return shallow<P, S>(comp);
}

export function toJson(wrapper: any) {
    return enzymetoJson(wrapper);
}

/**
 * TestIDを利用してComponetのPropsを取得します。
 *
 * @static
 * @template T Propsの型を指定
 * @param {ShallowWrapper} component
 * @param {string} name
 * @param {string} testId
 * @returns {T}
 * @memberof TestHelper
 */
export function getComponentPropsByTestID<T>(
    component: ShallowWrapper<any, any>,
    name: string,
    testId: string
): T | undefined {
    const findComponent = component.find(name).findWhere((wrapper: any) => {
        const targetProps = wrapper.props();
        return targetProps.testID === testId;
    });
    if (findComponent.length <= 0) {
        return undefined;
    }
    const props: T = findComponent.props();
    return props;
}

export function getButtonPropsByTestID(
    component: ShallowWrapper<any, any>,
    testID: string
): ButtonProps | undefined {
    return getComponentPropsByTestID<ButtonProps>(
        component,
        "MultiLingualButton",
        testID
    );
}

/*public static getTextBoxPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): ITextBoxProps | undefined {
        return this.getComponentPropsByTestID<ITextBoxProps>(
            component,
            "MultiLingualTextBox",
            testID
        );
    }

    public static getHeaderSearchBarPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IHeaderSearchBarProps | undefined {
        return this.getComponentPropsByTestID<IHeaderSearchBarProps>(
            component,
            "HeaderSearchBar",
            testID
        );
    }

    public static getListItemPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IListItemProps | undefined {
        return this.getComponentPropsByTestID<IListItemProps>(
            component,
            "MultiLingualListItem",
            testID
        );
    }*/

/*public static createNavigationMock(
        param: Partial<NavigationScreenProp<any, any>>
    ): NavigationScreenProp<any, any> {
        const navigation = new Object();
        Object.assign(navigation, param);
        return navigation as NavigationScreenProp<any, any>;
    }*/
