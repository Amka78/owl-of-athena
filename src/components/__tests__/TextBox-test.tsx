import "react-native";
import { ShallowWrapper } from "enzyme";
import React from "react";
import { TestHelper } from "../../utils/TestHelper";
import { Message } from "../../constants";
import { TextBox, TextBoxProps } from "../TextBox";
let component: ShallowWrapper<TextBoxProps, any, any>;
describe("TextBoxEx UnitTest", () => {
    it.each(["ja-JP", "en-US"])("renders correctly", (locale: string) => {
        Message.setLocale(locale);

        component = TestHelper.createMock(<TextBox></TextBox>);

        expect(TestHelper.toJson(component)).toMatchSnapshot();
    });
});
