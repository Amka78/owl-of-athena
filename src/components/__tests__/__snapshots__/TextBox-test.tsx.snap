// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBoxEx UnitTest renders correctly 1`] = `
<withTheme(TextInput)
  mode="flat"
  selectionColor="#ffffff"
  style={
    Object {
      "backgroundColor": "transparent",
      "fontFamily": "calibre_app_regular",
      "fontSize": 12,
      "marginBottom": 16,
      "width": 654,
    }
  }
  theme={
    Object {
      "colors": Object {
        "placeholder": "#ffffff",
        "primary": "#bcfeff",
        "text": "#bcfeff",
      },
    }
  }
  underlineColor="#ffffff"
/>
`;

exports[`TextBoxEx UnitTest renders correctly 2`] = `
<withTheme(TextInput)
  mode="flat"
  selectionColor="#ffffff"
  style={
    Object {
      "backgroundColor": "transparent",
      "fontFamily": "calibre_app_regular",
      "fontSize": 12,
      "marginBottom": 16,
      "width": 654,
    }
  }
  theme={
    Object {
      "colors": Object {
        "placeholder": "#ffffff",
        "primary": "#bcfeff",
        "text": "#bcfeff",
      },
    }
  }
  underlineColor="#ffffff"
/>
`;
